EIPS/eip-6780.md

```diff
@@ -27,37 +27,46 @@ The behaviour of `SELFDESTRUCT` is changed in the following way:
 
 1. When `SELFDESTRUCT` is executed in a transaction that is not the same as the contract calling `SELFDESTRUCT` was created:
 
- - `SELFDESTRUCT` does not delete any storage keys or code
- - `SELFDESTRUCT` transfers the entire account balance to the target.
- - No refund is given, as per [EIP-3529](./eip-3529.md).
- - [EIP-2929](./eip-2929.md)'s rules regarding `SELFDESTRUCT` remain unchanged.
+   - The current execution frame halts.
+   - `SELFDESTRUCT` does not delete any data (including storage keys, code, or the account itself).
+   - `SELFDESTRUCT` transfers the entire account balance to the target.
+   - Note that no refund is given since [EIP-3529](./eip-3529.md).
+   - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
   
 2. When `SELFDESTRUCT` is executed in the same transaction as the contract was created:
 
- - `SELFDESTRUCT` continues to behave as originally, i.e. deletes all storage keys and the account itself.
- - Subsequently, the account will behave like exactly like an empty account, both in the same transaction and in all later ones
- - Transfer the account balance to the target **and** set account balance to `0.`
- - Note that no refund is given since [EIP-3529](./eip-3529.md).
- - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
- - Note that when verkle tries are implemented on Ethereum, the cleared storage will be marked as having been written before but empty. This leads to no observable differences in EVM execution, but a contract having been created and deleted will lead to different state roots compared to the action not happening.
+   - `SELFDESTRUCT` continues to behave as originally, i.e. deletes all storage keys and the account itself, and execution halts.
+   - Subsequently, the account will behave exactly like an empty account, both in the same transaction and in all later ones.
+   - Transfer the account balance to the target **and** set account balance to `0`.
```

Comment:
I have a take about this:
Currently the self destruct does not deletes all storage and account instantly, but the account is registered to be destroyed, and will be at the end of the current transaction.
The described behavior seems that is intended to be instant. 
I would not recommend changing this particular behavior as it enables new ways of using it ( like deploy 2 different contracts in the same address on the same transaction, for example).

As i understand, the rationale is just enabling the current behavior if happens in the same transaction, but this also adds a little tweak to the current behavior.

Reply:
in that case i would add a clarification there about the new instant destruction^^

Reply:
I am not entirely sure about the intended behavior. the we implemented it is as described by @invocamanman that it will be registered for destruction at the end of the transaction. if this is not the intended behavior and instant destruction is required. then a clarification is definitely needed.

Reply:
Instant destruction is not required.  I updated the spec to make it clear with "`SELFDESTRUCT` continues to behave as it did prior to this EIP"